scipy/spatial/transform/rotation.py

```diff
@@ -1465,6 +1466,61 @@ def magnitude(self):
         else:
             return angles
 
+    def mean(self, weights=None, return_variance=False):
+        """Get the mean of the rotation(s).
+
+        Returns
+        -------
+        mean : `Rotation` instance
+            Object containing the mean of the rotations in the current
+            instance.
+        weights : array_like shape (N,), optional
+            Weights describing the relative importance of the rotations. If
+            None (default), then all values in `weights` are assumed to be
+            equal.
+        return_variance : bool, optional
+            Whether to return the variance of the rotation(s). Default is
+            False.
+
+        Notes
+        -----
+        The mean used is the chordal L2 mean (also called the projected or
+        induced arithmetic mean).
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> r = R.random(10, random_state=0)
```

Comment:
I think this example is not very meaningful, moreover the resulting mean quaternion can be returned multiplied by -1 and the doctest will fail (this happened).

I suggest generating a set of rotation through Euler angles filling some narrow ranges uniformly. And then represent the mean as Euler angles too. This way we can see that the average has angles somewhere in the middle.

Reply:
I have added a very simple example along the lines you suggest.